mesheryctl/pkg/utils/auth.go

```diff
@@ -58,6 +60,28 @@ func NewRequest(method string, url string, body io.Reader) (*http.Request, error
 		return nil, ErrAttachAuthToken(err)
 	}
 
+	// check status code from request, checks for issues with auth token
+	resp, err := client.Do(req)
+
+	if err != nil && resp == nil {
+		return req, ErrFailRequest(err)
+	}
+
+	// failsafe for having an expired token
+	if resp.StatusCode == 302 {
+		return req, InvalidToken()
+	}
+
+	// failsafe for not being authenticated
+	if ContentTypeIsHTML(resp) {
+		return req, ErrUnauthenticated()
+	}
+
+	// failsafe for bad api call
+	if resp.StatusCode != 200 {
+		return req, ErrFailReqStatus(resp.StatusCode)
+	}
+
 	return req, nil
```

Comment:
```suggestion
	return resp, nil
```

Now that we executed request inside the function itself, let;s return the response back, instead of returning request.

Reply:
if NewRequest returns a http response now, then how should we handle something like [this](https://github.com/meshery/meshery/blob/048517fa89c3028fd922af6abea5af2a0cb7a2d1/mesheryctl/internal/cli/root/perf/apply.go#L217-L222) where we need a request returned?

Reply:
Nevermind I see, http.NewRequest function should work correct?

Reply:
actually there are a few different instances where a request is expected from utils.NewRequest. Is it safe to say that anywhere that request is made http.NewRequest can be called instead?

Reply:
yes, all those places will need a little cleaning, the function will directly returns the response fetched, and then that response can be used further and also closed with `defer` after getting the work done.

Reply:
Nice @cpepper96 I completely forgot about such instances. I think we should leave it here then, now that the function makes sure token isn't expired, and then we make an actual request sounds also okay to me. We can go one step ahead and make this function more complicated but that just sounds like over-work to me. 

but now that we are making a "pre-request" to confirm our token condition, then in cases like `/profileID/run` it is going to trigger a performance test without those query parameters attached.

CC @navendu-pottekkat @hexxdump @leecalcote

Reply:
Maybe we should make this "pre-request" to an API which doesn't change or trigger anything on backend but can confirm our token condition.

The API endpoint `/api/user/token` sounds good to me, It returns the token as file but also needs an attached token in the request. This endpoint looks good to me that can act as "pre-request" to confirm our token condition.

Reply:
the attempt to centralize these expired and invalid token checks has lead us in the hole of "pre-request" and making an request after that again.

We need a little discussion on this again, I'm out of ideas rn, I'll think about what can this be handled better soon.